src/main/java/uk/co/onsdigital/discovery/metadata/api/model/Dimension.java

```diff
@@ -1,75 +1,132 @@
 package uk.co.onsdigital.discovery.metadata.api.model;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.google.common.annotations.VisibleForTesting;
+import uk.co.onsdigital.discovery.model.DimensionValue;
+import uk.co.onsdigital.discovery.model.DimensionalDataSet;
+import uk.co.onsdigital.discovery.model.Hierarchy;
 
-import java.util.Collections;
+import javax.persistence.*;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Objects;
-import java.util.Set;
+import java.util.UUID;
 
 /**
- * Represents metadata about a dimension of a dataset.
+ * Represents a dimension in a dataset.
  */
-@JsonInclude(JsonInclude.Include.NON_EMPTY)
-public class Dimension implements Comparable<Dimension> {
-    private String id;
+@Entity
+@Table(name = "dimension")
+@IdClass(Dimension.DimensionPK.class)
+@NamedQueries({
+        @NamedQuery(name = "Dimension.findByDataSetId", query = "SELECT d FROM Dimension d LEFT JOIN FETCH d.hierarchy WHERE d.dataSet.id = :dataSetId")
+})
+public class Dimension {
+
+    @Id
+    @ManyToOne
+    @JoinColumn(name = "dimensional_data_set_id", columnDefinition = "uuid")
+    private DimensionalDataSet dataSet;
+
+    @Id
+    @Column(name = "name")
     private String name;
+
+    @OneToMany
+    @JoinColumns({
+            @JoinColumn(name = "dimensional_data_set_id", referencedColumnName = "dimensional_data_set_id", columnDefinition = "uuid", insertable = false, updatable = false),
+            @JoinColumn(name = "name", referencedColumnName = "name", insertable = false, updatable = false)
+    })
+    private List<DimensionValue> values;
+
+    @ManyToOne
+    @JoinColumn(name = "hierarchy_id", referencedColumnName = "id", columnDefinition = "uuid", insertable = false, updatable = false)
+    private Hierarchy hierarchy;
+
+    @Transient
     private String url;
-    private Set<DimensionOption> options = Collections.emptySet();
 
-    public String getId() {
-        return id;
+    @Transient
```

Comment:
This class should be split in two: one for persistence (jpa), another for the api (json).
This will be particularly important when we introduce db change management - we'll want all db-related classes/sql in the backend model.